src/tests/utests/iotc_utest_bsp_crypto_ecc.cc

```diff
@@ -0,0 +1,259 @@
+};
+
+/**
+ * Verify an EC signature using OpenSSL.
+ *
+ * @param hash the hash that was signed.
+ * @param hash_len length of the hash in bytes.
+ * @param sig signature to verify, formated according to rfc7518 section 3.4.
+ * @param sig_len length of the signature in bytes.
+ * @param pub_key_pem public key in PEM format.
+ * @returns 0 if the verification succeeds.
+ */
+int IotcBspCryptoEcc ::ec_verify_openssl(const uint8_t *hash, size_t hash_len,
+                                         const uint8_t *sig, size_t sig_len,
+                                         const char *pub_key_pem) {
+  if (sig_len != 64) {
+    iotc_debug_format("sig_len expected to be 64, was %d", sig_len);
+    return -1;
+  }
+
+  int ret = -1;
+  EC_KEY *ec_key_openssl = nullptr;
+  ECDSA_SIG *sig_openssl = nullptr;
+
+  // Parse the signature to OpenSSL representation
+  constexpr int kIntLength = 32; // as per RFC7518 section-3.4
+  BIGNUM *r = BN_bin2bn(sig, kIntLength, nullptr);
+  BIGNUM *s = BN_bin2bn(sig + kIntLength, kIntLength, nullptr);
+  sig_openssl = ECDSA_SIG_new();
+
+  // Parse the public key to OpenSSL representation
+  BIO *pub_key_pem_bio = BIO_new(BIO_s_mem());
+  BIO_puts(pub_key_pem_bio, pub_key_pem);
+  ec_key_openssl =
+      PEM_read_bio_EC_PUBKEY(pub_key_pem_bio, nullptr, nullptr, nullptr);
+  if (ec_key_openssl == nullptr) {
+    goto cleanup;
+  }
+
+// 1.1.0+, see opensslv.h for versions
+#if 0x10100000L <= OPENSSL_VERSION_NUMBER
+  if (1 != ECDSA_SIG_set0(sig_openssl, r, s)) {
+    goto cleanup;
+  }
+#else // Travis CI's 14.04 trusty supports openssl 1.0.1f
+  sig_openssl->r = r;
+  sig_openssl->s = s;
+#endif
+
+  if (1 == ECDSA_do_verify((const unsigned char *)hash, hash_len, sig_openssl,
+                           ec_key_openssl)) {
+    ret = 0; // Verify success
+  }
+
+cleanup:
+  ERR_print_errors_fp(stderr);
+  EC_KEY_free(ec_key_openssl);
+  ECDSA_SIG_free(sig_openssl);
+  BIO_free_all(pub_key_pem_bio);
+
+  return ret;
+}
+
+TEST_F(IotcBspCryptoEcc, SmallBuffer) {
+  constexpr size_t kEccSignatureBufferLength = 1;
+  uint8_t ecc_signature[kEccSignatureBufferLength];
+  size_t bytes_written = 0;
+  EXPECT_EQ(iotc_bsp_ecc(&DefaultPrivateKey, ecc_signature,
+                         kEccSignatureBufferLength, &bytes_written,
+                         kDefaultDataToSign, kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_BUFFER_TOO_SMALL_ERROR);
+}
+
+TEST_F(IotcBspCryptoEcc, SmallBufferProperMinSize) {
+  constexpr size_t kEccSignatureBufferLength = 1;
+  uint8_t ecc_signature[kEccSignatureBufferLength];
+  uint8_t ecc_signature_new[64];
+  size_t bytes_written = 0;
+
+  EXPECT_EQ(iotc_bsp_ecc(&DefaultPrivateKey, ecc_signature,
+                         kEccSignatureBufferLength, &bytes_written,
+                         kDefaultDataToSign, kDefaultDataToSignLength),
+            IOTC_BSP_CRYPTO_BUFFER_TOO_SMALL_ERROR);
+  // Two 32-byte integers(r and s) build up a JWT ECC signature
```

Comment:
signature.

Use proper sentences and punctuation for comments. Otherwise the reader might have to think whether this comment was mistakenly truncated, or it's all what you had to say.